apps/frontend/src/app/PageCharacter/CharacterDisplay/Tabs/TabOptimize/Components/AllowChar.tsx

```diff
@@ -0,0 +1,111 @@
+import ModalWrapper from "../../../../../Components/ModalWrapper";
+import SqBadge from "../../../../../Components/SqBadge";
+import { CharacterContext } from "../../../../../Context/CharacterContext";
+import { DatabaseContext } from "../../../../../Database/Database";
+import useBoolState from "../../../../../ReactHooks/useBoolState";
+import useForceUpdate from "../../../../../ReactHooks/useForceUpdate";
+import { toggleArr } from "../../../../../Util/Util";
+import useBuildSetting from "../useBuildSetting";
+export default function AllowChar({ disabled = false, numArtsEquippedUsed }: { disabled?: boolean, numArtsEquippedUsed: number }) {
+  const { t } = useTranslation("page_character_optimize")
+  const { character: { key: characterKey } } = useContext(CharacterContext)
+  const { buildSetting: { allowLocations }, buildSettingDispatch } = useBuildSetting(characterKey)
+  const { database } = useContext(DatabaseContext)
+  const [show, onOpen, onClose] = useBoolState(false)
+  const [dbDirty, forceUpdate] = useForceUpdate()
+  const deferredDbDirty = useDeferredValue(dbDirty)
+
+  useEffect(() => database.charMeta.followAny(s => forceUpdate()), [forceUpdate, database])
+  useEffect(() => database.chars.followAny(s => forceUpdate()), [forceUpdate, database])
+
+  const locList = useMemo(() => deferredDbDirty && Array.from(new Set(
+    Object.entries(database.chars.data)
+      .filter(([ck]) => ck !== characterKey)
+      .sort(([ck1, c1], [ck2, c2]) => {
+        // sort characters by: star => more artifacts equipped
+        const [choosec1, choosec2] = [-1, 1]
+        const c1f = database.charMeta.get(ck1).favorite
+        const c2f = database.charMeta.get(ck2).favorite
+        if (c1f && !c2f) return choosec1
+        else if (c2f && !c1f) return choosec2
+
+        const art1 = Object.values(c1.equippedArtifacts).filter(id => id).length
+        const art2 = Object.values(c2.equippedArtifacts).filter(id => id).length
+        if (art1 > art2) return choosec1
+        else if (art2 > art1) return choosec2
+        return ck1.localeCompare(ck2)
+
+      }).map(([k]) => charKeyToLocCharKey(k))))
+    , [deferredDbDirty, characterKey, database])
+
+  const allowAll = useCallback(() => buildSettingDispatch({ allowLocations: [...locList] }), [buildSettingDispatch, locList])
+  const disallowAll = useCallback(() => buildSettingDispatch({ allowLocations: [] }), [buildSettingDispatch])
+
+  const toggle = useCallback((lk: LocationKey) => buildSettingDispatch({ allowLocations: toggleArr(allowLocations, lk) }), [allowLocations, buildSettingDispatch])
+
+  const total = locList.length
+  const useTot = allowLocations.length
+  return <Box display="flex" gap={1}>
+    <ModalWrapper open={show} onClose={onClose} containerProps={{ maxWidth: "xl" }}><CardDark>
+      <CardContent>
+        <Box display="flex" gap={1} >
+          <Typography variant="h6" flexGrow={1}>{t`excludeChar.modal.title_sel`}</Typography>
```

Comment:
```suggestion
          <Typography variant="h6" flexGrow={1}>{t`excludeChar.modal.title`}</Typography>
```

Reply:
Do you want to add a tooltip like for Exclude Artifacts? Maybe its simple enough, just pointing out any differences between the two

Reply:
should be pretty self-explanatory